src/validator.rs

```diff
@@ -749,26 +713,22 @@ impl<'a> Validator<'a> {
 
         if ContentMatchPair::contains_mismatch(&url_match_pairs) {
             info!("Link destination Error");
-            Ok(Some(vec![
-                Box::new(ContentError::new(
-                    rx,
-                    node,
-                    &self.prescription,
-                    &self.document,
-                    url_match_pairs.clone(),
-                )?),
-            ]))
+            Ok(Some(vec![Box::new(ContentError::new(
+                rx,
+                node,
+                &self.prescription,
+                &self.document,
+                url_match_pairs.clone(),
+            )?)]))
```

Comment:
This same block is repeated a bunch of times; can you think of any way to tidy it up a bit? Maybe with a little anonymous fn; not sure if that's a thing.